clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -847,7 +845,33 @@ template <class Emitter>
 bool ByteCodeExprGen<Emitter>::VisitOpaqueValueExpr(const OpaqueValueExpr *E) {
   if (Initializing)
     return this->visitInitializer(E->getSourceExpr());
-  return this->visit(E->getSourceExpr());
+
+  PrimType CacheVariableTy = classify(E).value_or(PT_Ptr);
```

Comment:
Do you have a test case for the non-primitive code path here? If not, we should rather use `classifyPrim` and handle it when we see it in the wild.

Reply:
I added a test case like that.

```
constexpr int foo() { return X(0) ?: X(0); }
```
```
`-OpaqueValueExpr ... 'X':'X' xvalue
   `-MaterializeTemporaryExpr ... 'X':'X' xvalue
     `-CXXFunctionalCastExpr ... 'X':'X' functional cast to X <NoOp>
       `-CXXParenListInitExpr ... 'X':'X'
         `-IntegerLiteral ... 'int' 0
```